packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -410,106 +869,777 @@ class _CupertinoAlertActionSection extends StatelessWidget {
+    }
+
+    if (!enabled) {
+      style = style.copyWith(color: style.color.withOpacity(0.5));
     }
+
+    // Apply a sizing policy to the action button's content based on whether or
+    // not the device is in accessibility mode.
+    // TODO(mattcarroll): The following logic is not entirely correct. It is also
+    // TODO(mattcarroll): the case that if content text does not contain a space,
+    // TODO(mattcarroll): it should also wrap instead of ellipsizing. We are
+    // TODO(mattcarroll): consciously not implementing that now due to complexity.
+    final Widget sizedContent = _CupertinoAccessibility.of(context)?.isInAccessibilityMode ?? false
+      ? _buildContentWithAccessibilitySizingPolicy(
+          textStyle: style,
+          content: child,
+        )
+      : _buildContentWithRegularSizingPolicy(
+          context: context,
+          textStyle: style,
+          content: child,
+        );
+
+    return new GestureDetector(
+      onTap: onPressed,
+      behavior: HitTestBehavior.opaque,
+      child: new ConstrainedBox(
+        constraints: const BoxConstraints(
+          minHeight: _kMinButtonHeight,
+        ),
+        child: new Container(
+          alignment: Alignment.center,
+          padding: new EdgeInsets.all(_calculatePadding(context)),
+          child: sizedContent,
+        ),
+      ),
+    );
   }
 }
 
-// A CustomPainter to draw the divider lines.
+// iOS style dialog action button layout.
+//
+// [_CupertinoDialogActionsRenderWidget] does not provide any scrolling
+// behavior for its buttons. It only handles the sizing and layout of buttons.
+// Scrolling behavior can be composed on top of this widget, if desired.
 //
-// Draws the cross-axis divider lines, used when the layout is horizontal.
-class _CupertinoHorizontalDividerPainter extends CustomPainter {
-  _CupertinoHorizontalDividerPainter(this.count);
+// See [_RenderCupertinoDialogActions] for specific layout policy details.
+class _CupertinoDialogActionsRenderWidget extends MultiChildRenderObjectWidget {
+  _CupertinoDialogActionsRenderWidget({
+    Key key,
+    @required List<Widget> actionButtons,
+    double dividerThickness = 0.0,
+  }) : _dividerThickness = dividerThickness,
+       super(key: key, children: actionButtons);
 
-  final int count;
+  final double _dividerThickness;
 
   @override
-  void paint(Canvas canvas, Size size) {
-    final Paint paint = new Paint()..color = _kButtonDividerColor;
-
-    canvas.drawLine(Offset.zero, new Offset(size.width, 0.0), paint);
-    for (int i = 1; i < count; ++i) {
-      // TODO(abarth): Hide the divider when one of the adjacent buttons is
-      // highlighted.
-      final double x = size.width * i / count;
-      canvas.drawLine(new Offset(x, 0.0), new Offset(x, size.height), paint);
-    }
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderCupertinoDialogActions(
+      dialogWidth: _CupertinoAccessibility.of(context)?.isInAccessibilityMode ?? false
```

Comment:
Same here, I suspect if this widget is built once and then the text scale changes on the fly, this dialog width wouldn't update since it goes through updateRenderObject

Reply:
added dialog width to the update call